packages/cli/src/index.ts

```diff
@@ -34,9 +46,31 @@ process.on('SIGTERM', handleSigTerm);
           .default('auto', 'detect package manager automatically')
       )
       .description('replace redundant polyfills inside your project\'s node_modules with nolyfill')
-      .action((source: string | undefined, option: unknown) => {
-        console.log(source, option);
-        throw new Error('not implemented');
+      .action(async (source: string | undefined, option: PmCommandOptions) => {
+        const projectPath = path.resolve(source ?? process.cwd());
+        const packageManager = option.pm === 'auto' ? await detectPackageManager(projectPath) : option.pm;
+
+        const searchResult = await searchPackages(packageManager, projectPath, allPackages);
+        // console.log(renderTree(searchResult));
+
+        const leafNodes = getDedupeLeafNodes(searchResult);
+        console.log('The following packages will be overridden:');
+        console.log(renderTree(leafNodes));
+
+        const overrides = Object.fromEntries(leafNodes.map((node) => [
+          node.name,
+          `npm:@nolyfill/${node.name}@latest`
+        ]));
+        const packageJsonPath = path.join(projectPath, 'package.json');
+        const packageJson = await fs.readFile(packageJsonPath, 'utf-8');
+        const parsedPackageJson = JSON.parse(packageJson);
+        parsedPackageJson.overrides = {
+          ...parsedPackageJson.overrides,
+          ...overrides
+        };
+        // TODO: respect existing format?
+        // TODO: confirm with user
+        await fs.writeFile(packageJsonPath, JSON.stringify(parsedPackageJson, null, 2));
```

Comment:
pnpm uses `.pnpm.overrides`, not `.overrides`: https://pnpm.io/package_json#pnpmoverrides

yarn uses `.resolutions`: https://yarnpkg.com/configuration/manifest#resolutions

Need to handle them separately.

Reply:
updated